### DISCUSSION TITLE
Import from Foam

### DISCUSSION THREAD

**Discussion by TheOneValen** [TriliumNext/Trilium#3420] [Q&A] [answered]
Hi!

I tried trilium and it looks very promising. But I already have a big archive of notes made with foam. This is basically just a bunch of markdown files. I tried importing the notes using the import function, but there are two things that did not work and would be needed for me to be able to migrate:

All links between notes are wikilink `[[another note title or filename]]` style. No links are generated on import into trilium.

The journal is a bunch of markdown files following a specific pattern `YYY-MM-DD`.

Is there a way I can import the links and the journal structure? Is there something I can do to prepare the files for import (e.g. some regex foo)?

Thank you!

P.S. Extracting `#tag` like tags is also important.

**Comment by zadam** [comment, marked as answer]
For link preservation, you might try to convert first to HTML using some converter which understands Foam syntax.

For tags and journal notes I guess you would have to create some import script using [ETAPI](github.com/zadam/trilium/wiki/ETAPI).

**Reply by TheOneValen** [reply]
Thank you. converting `[[]]` links to `[]()` Markdown links actually works very well. So for the journal I would have to put the Attribute of each day on the corresponding note, right? I can skip the template attribute for the older stuff?

**Reply by zadam** [reply]
Only `#dateNote` label is necessary to be recognized as a journal note for the given note. But it will not be automatically added to the journal hierarchy - if you want to keep that tree structure for older notes then you will have to add those notes into the tree structure in the import script.

**Reply by TheOneValen** [reply]
Worked like a charm. Made a script to replace `[[]]` links with relative markdown links. A second script for adding the labels to the journal files. I can upload them somewhere if anybody is interested.

**Reply by TheOneValen** [reply]
Foam had the abillity to type `//` and then a quickselect opened for a date, then the journal entry of the date was linked.

Is there a way I can configure a keyboard shortcut, that creates a link to the current daily journal (or opens a list of journal entry today-+7 days?)
